apps/functions-subscription/src/domain/subscription.ts

```diff
@@ -0,0 +1,53 @@
+import { pipe } from 'fp-ts/lib/function';
+import * as O from 'fp-ts/Option';
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import * as TE from 'fp-ts/TaskEither';
+import { Capabilities } from './capabilities';
+
+export type SubscriptionId = string & { readonly __tag: unique symbol };
+export type UserId = string & { readonly __tag: unique symbol };
+export type TrialId = string & { readonly __tag: unique symbol };
```

Comment:
what's the purpose of `__tag` ?

Reply:
Without it, these three IDs are interchangeable, so we can call, for example, `makeSubscriptionId` with the parameters in any order. We want to prevent this kind of error at compile time:
```ts
// no compile error
makeSubscriptionId(trialId, userId);

// raise a compile error
makeSubscriptionId(userId, trialId);
```